pkg/resource/unstructured/composite/composite.go

```diff
@@ -154,6 +154,28 @@ func (c *Unstructured) SetReclaimPolicy(p v1alpha1.ReclaimPolicy) {
 	_ = fieldpath.Pave(c.Object).SetValue("spec.reclaimPolicy", p)
 }
 
+// SetComposedResourceCount of this Composite resource.
+func (c *Unstructured) SetComposedResourceCount(i int) {
+	_ = fieldpath.Pave(c.Object).SetNumber("status.composedResources", float64(i))
+}
```

Comment:
This is a broader question, but is the reason why we have to treat these new object types as unstructured rather than a type that satisfies an interface because we start controllers for them without having representations of their actual types? For instance, in the managed reconciler we have a go struct that we pass into the `NewReconciler` function, but here we are starting controllers on the fly for arbitrary object types. However, we still know they can satisfy these interfaces because they were produced by other controllers that we write. Do I have that correct?